subworkflows/local/deepvariant_caller.nf

```diff
@@ -1,41 +1,66 @@
 //
-// Check input samplesheet and get read channels
+// call deepvariant
 //
 
-include { SAMTOOLS_VIEW } from '../../modules/nf-core/samtools/view/main'
-include { DEEPVARIANT }   from '../../modules/nf-core/deepvariant/main'
+include { DEEPVARIANT }         from '../../modules/nf-core/deepvariant/main'
+include { BCFTOOLS_CONCAT as BCFTOOLS_CONCAT_VCF   }   from '../../modules/nf-core/bcftools/concat/main'
+include { BCFTOOLS_CONCAT as BCFTOOLS_CONCAT_GVCF  }   from '../../modules/nf-core/bcftools/concat/main'
 
 workflow DEEPVARIANT_CALLER {
 
     take:
-    reads    // [ val(meta), data, index ]
-    fasta    // path to reference fasta file, either compressed or uncompressed
-    fai      // path to compressed or uncompressed reference fasta index file
-    gzi      // path to compressed fasta index file
-    interval // path to interval bed file
+    reads_fasta    // [ val(meta), cram, crai, interval, fasta_file_name, fasta, fai ]
 
     main:
     ch_versions = Channel.empty()
 
-    SAMTOOLS_VIEW ( reads, fasta, [] )
-    ch_versions = ch_versions.mix ( SAMTOOLS_VIEW.out.versions )
-     
-    cram_crai = SAMTOOLS_VIEW.out.cram
-                .join(SAMTOOLS_VIEW.out.crai)
-                .map { filtered_reads -> filtered_reads + [interval ?: []] }
+    reads_fasta.map{ meta, cram, crai, interval, fasta_file_name, fasta, fai ->
+                    [ [ id: meta.id + "_" + fasta_file_name, sample: meta.id, type: meta.type ], 
+                       cram,
+                       crai, 
+                       interval 
+                    ] }
+               .set{ cram_crai } // [ val(meta), cram, crai, interval ]
 
-    if ( !gzi ) {
-        gzi = []
-    }
+    // fasta
+    reads_fasta.map{ meta, cram, crai, interval, fasta_file_name, fasta, fai -> [ fasta ] }
+            .set{ fasta }
 
+    // fai
+    reads_fasta.map{ meta, cram, crai, interval, fasta_file_name, fasta, fai -> [ fai ] }
+            .set{ fai }
+
+    // split fasta in compressed format, no gzi index file needed
+    gzi = []
```

Comment:
What happens if someone provides the `gzi` file?

Reply:
This gzi file for the split fasta files, not for the original input fasta files.

input gzi file still needed if the given fasta file is compressed, that will be used by samtools view filtering.